sdk/core/Azure.Core.TestFramework/src/ClientDiagnosticListener.cs

```diff
@@ -123,7 +122,7 @@ public void Dispose()
             }
 
             List<IDisposable> subscriptions;
-            lock (_subscriptions)
+            lock (Scopes)
```

Comment:
Why lock on `Scopes` here? We lock on `_subscriptions` in OnNext (or at least what it used to, if changed concurrently) so we should lock on it here again to synchronize.

Reply:
Good point, I'll change that one too. I want to synchronize dispose and `OnNext(KeyValuePair<string, object> value)`